README.md

```diff
@@ -1,130 +1,182 @@
+
+### Usage
+
+### Import `sdk.js`
+
+Import `sdk.js` on your client. Call the `Smartcar` constructor then use one of its instance methods to setup the authorization flow.
 
-### Import `sdk.js`:
 ```html
-<script src="https://cdn.smartcar.com/javascript-sdk/sdk-1.0.0.js"></script>
+<script src="https://javascript-sdk.smartcar.com/sdk-2.0.0.js"></script>
 ```
 
-### Use `sdk.js`:
+### Use `sdk.js`
+
 Initialize:
+
 ```javascript
 const smartcar = new Smartcar({
   clientId: `your-client-id`,
   redirectUri: `your-redirect-uri`,
-  scope: ['read_vehicle_info', 'read_odometer'], // optional
-  onComplete: function() { // optional
-    window.location.reload();
+  // optional, defaults to all scopes
+  scope: ['read_vehicle_info', 'read_odometer'],
+  // if using smartcar hosting this is required and must take at least two
+  // arguments - error & code
+  // otherwise optional (will be called with the same arguments but is not
+  // required to handle them)
+  onComplete: function(error, code, state <optional>) {
+    // actions to take on completion of auth flow
+    // if using smartcar hosting this should send the code to your backend
+    // server to exchange for an access token
   },
-  development: false, // optional
+  development: false, // optional, defaults to false
+  useSmartcarHostedRedirect: true, // optional, defaults to false
 });
 ```
-The best placement for the above code is just before the closing `</body>` tag.
 
-To open the Smartcar OAuth dialog directly:
-```javascript
-// note: state and forcePrompt are optional
-smartcar.openDialog({state, forcePrompt});
-```
+<!-- TODO: why show state and forcePrompt in this? just seems to add unnecessary disclaimers -->
 
-To add a click handler to an HTML element that will open the Smartcar OAuth dialog:
-```javascript
-// note: state and forcePrompt are optional
-smartcar.addClickHandler({id, state, forcePrompt});
-```
+Once initialized there are three instance methods for setting up the OAuth flow:
 
-You may embed `callback.js` in the page that is served by your redirectUri. Upon loading, this file will invoke the `onComplete` callback specified above and close the window:
-```html
-<script src="https://cdn.smartcar.com/javascript-sdk/callback-1.0.0.js"></script>
-```
+- `openDialog()` - to open the Smartcar OAuth dialog directly
+- `addClickHandler({id})` - to add a click handler to an HTML element that will open the Smartcar OAuth dialog
+- `generateLink()` - to generate the bare Smartcar OAuth URL
 
-To retrieve the Smartcar OAuth URL (useful in single page applications when redirecting to Smartcar's OAuth flow in a new tab):
-```javascript
-// note: state and forcePrompt are optional
-smartcar.generateLink({state, forcePrompt});
-```
+See the [reference below](TODO: link to appropriate section) for detailed signature information.
 
 ## Installation
+
 You can import this SDK into your application from Smartcar's CDN:
 
 ### sdk.js
+
```

Comment:
What's the difference between this and the one used above (in the `import sdk.js` section)?

Also, should we place installation before usage?

Reply:
Good question, they shouldn't both need to exist. Happy to place installation before usage if we have consensus, @sankethkatta any thoughts?

Reply:
I'm fine leaving it as is for now and re-arranging afterwards if we feel the organization could be better. Right now, I think it flows alright and matches with the old structure.